prisma/migrations/20220906193601_add_initial_models/migration.sql

```diff
@@ -0,0 +1,106 @@
+-- CreateEnum
+CREATE TYPE "RoleType" AS ENUM ('WINTER_INTERNSHIP', 'SPRING_INTERNSHIP', 'SUMMER_INTERNSHIP', 'FALL_INTERNSHIP', 'FULL_TIME');
+
+-- CreateEnum
+CREATE TYPE "ApplicationStageType" AS ENUM ('APPLIED', 'ONLINE_ASSESSMENT', 'TECHNICAL', 'NON_TECHNICAL', 'MIXED', 'OFFERED', 'ACCEPTED', 'REJECTED', 'WITHDRAWN');
+
+-- CreateTable
+CREATE TABLE "users" (
+    "uid" TEXT NOT NULL,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "users_pkey" PRIMARY KEY ("uid")
+);
+
+-- CreateTable
+CREATE TABLE "companies" (
+    "id" SERIAL NOT NULL,
+    "name" TEXT NOT NULL,
+    "company_url" TEXT NOT NULL,
+    "is_verified" BOOLEAN NOT NULL DEFAULT false,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "companies_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "roles" (
+    "id" SERIAL NOT NULL,
+    "company_id" INTEGER NOT NULL,
+    "title" TEXT NOT NULL,
+    "type" "RoleType" NOT NULL,
+    "year" INTEGER NOT NULL,
+    "is_verified" BOOLEAN NOT NULL DEFAULT false,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "roles_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "applications" (
+    "id" SERIAL NOT NULL,
+    "user_id" TEXT NOT NULL,
+    "role_id" INTEGER NOT NULL,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "applications_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "application_stages" (
+    "id" SERIAL NOT NULL,
+    "application_id" INTEGER NOT NULL,
+    "type" "ApplicationStageType" NOT NULL,
+    "date" TIMESTAMP(3) NOT NULL,
+    "reaction_id" INTEGER,
+    "updated_at" TIMESTAMP(3) NOT NULL,
+
+    CONSTRAINT "application_stages_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "reactions" (
+    "id" SERIAL NOT NULL,
+    "emoji_unicode_hex" TEXT NOT NULL,
+    "remark" TEXT NOT NULL,
```

Comment:
Must users always add a remark? Is it possible to put an emoji without a remark?

Reply:
I would prefer forcing empty string over null for text and I can see how some users want to write text without emojis.

Reply:
I thought that the empty string would be fine, but on second thought perhaps `null` would be better.

Changed to make it optional.

Reply:
Oops didn't see @bharathcs 's comment when I wrote the reply, but I am open for discussion.

Reply:
I think if either can be null, and there is at most 1 reaction per stage, then we can just put these columns into application_stages and do without a reactions table.

Reply:
On nullity: My thinking was that we will not have different behaviour for `null` vs `''`, but I'm not opposed to allowing nulls, it only means there's an additional check in the client side.

On reactions: I agree with removing the reactions table and adding nullable columns. The downside is that its less extensible if we want to expand to do more than just a simple text box (e.g. additional records / images etc). I think that's a very minor reason though, and to avoid complexity we should just remove the table and simplify the design.

Reply:
Removed reactions table and moved fields into `application_stages` table in commit https://github.com/sivayogasubramanian/caza/pull/9/commits/1660732a9d67ada754e80d624bb62d215c162237